dbt/contracts/connection.py

```diff
@@ -98,6 +104,9 @@
         'password': {
             'type': 'string',
         },
+        'authenticator': {
+            'type': 'string'
+        },
```

Comment:
based on https://docs.snowflake.net/manuals/user-guide/admin-security-fed-auth-use.html#browser-based-sso it looks like this can be one of `externalbrowser` or a valid Okta URL. can you add a description here explaining that? we'll incorporate it into docs.getdbt.com later on

Reply:
Let me know if that's good, wasn't sure if I should put an actual link to the Snowflake connector doc in the description